src/components/CodingAgentApp.tsx

```
@@ -499,20 +510,22 @@ export default function CodingAgentApp() {
+   */
   useEffect(() => {
-    const id = pendingJump.current;
-    if (!id) return;
-    const row = rowFor(runsList.current, id);
-    if (!row) return;
-    pendingJump.current = null;
-    // jsdom has no scrollIntoView; every browser does.
-    row.scrollIntoView?.({ block: "nearest" });
-  }, [runsShown, expanded]);
+    const pending = takePendingCodingRun();
+    if (pending) { setOpenRunId(pending); setPage("home"); }
```

ESLint

[error] Error: Calling setState synchronously within an effect can trigger cascading renders

Effects are intended to synchronize state between React and external systems such as manually updating the DOM, state management libraries, or other platform APIs. In general, the body of an effect should do one or both of the following:
* Update external systems with the latest state from React.
* Subscribe for updates from some external system, calling setState in a callback function when external state changes.

Calling setState synchronously within an effect body causes cascading renders that can hurt performance, and is not recommended. (https://react.dev/learn/you-might-not-need-an-effect).

/home/jailuser/git/src/components/CodingAgentApp.tsx:521:20
  519 |   useEffect(() => {
  520 |     const pending = takePendingCodingRun();
> 521 |     if (pending) { setOpenRunId(pending); setPage("home"); }
      |                    ^^^^^^^^^^^^ Avoid calling setState() directly within an effect
  522 |     const handler = (e: Event) => {
  523 |       const id = (e as CustomEvent<{ runId?: unknown }>).detail?.runId;
  524 |       if (typeof id === "string" && id) { setOpenRunId(id); setPage("home"); }

(react-hooks/set-state-in-effect)